### ISSUE TITLE
HLSL: Order of const keyword can break parsing

### ISSUE THREAD

**Issue by TheJackiMonster** [KhronosGroup/glslang#3092]
While trying to compile some HLSL shaders I encountered the following issue with the `const` keyword using for local variables and function arguments.

If you use following syntax, everything is fine:
```
const float f = 1.0f;
```

But using the following syntax will throw an error by the HLSL parser:
```
float const f = 1.0f;
```

Unfortunately some shaders are written by people crosing both variants. ^^'

Another example how the current parser seems to evaluate code with functions:

GOOD:
```
void test(const float3 f) {}
```

BAD:
```
void test(float3 const f) {}
```

**Comment by greg-lunarg** [comment]
Yes, this is an issue. dxc accepts `float const f = 1.0f;`

**Comment by arcady-lunarg** [comment]
Fixed by #3256.
